jetty-core/jetty-io/src/main/java/org/eclipse/jetty/io/Content.java

```diff
@@ -866,6 +908,36 @@ static void write(Sink sink, boolean last, String utf8Content, Callback callback
             sink.write(last, ByteBuffer.wrap(utf8Content.getBytes(StandardCharsets.UTF_8)), callback);
         }
 
+        static void write(Sink sink, boolean last, Content.Source source, Callback callback)
+        {
+            Content.Source.Aware aware = findContentSourceAware(sink);
+            if (aware != null)
+            {
+                // Optimization to enable zero-copy.
+                aware.setContentSource(source);
+                sink.write(last, TRANSFER, callback);
+            }
+            else
+            {
+                // Normal source.read() + sink.write() full copy.
+                Content.copy(source, last, sink, callback);
+            }
+        }
+
+        private static Content.Source.Aware findContentSourceAware(Sink sink)
+        {
+            while (true)
+            {
+                if (sink instanceof Content.Source.Aware aware)
+                    return aware;
+                if (sink instanceof Wrapper wrapper)
```

Comment:
I think if the sink is wrapped then we should not bypass it with transferTo
Perhaps we need an AwareWrapper, that would allow itself to be bypassed?